yarn-project/aztec.js/src/account_manager/deploy_account_method.ts

```diff
@@ -0,0 +1,137 @@
+import { type FunctionCall, type PXE, PackedArguments, type Tx, TxExecutionRequest } from '@aztec/circuit-types';
+import {
+  AztecAddress,
+  type EthAddress,
+  type Fr,
+  FunctionData,
+  type PublicKey,
+  TxContext,
+  getContractInstanceFromDeployParams,
+} from '@aztec/circuits.js';
+import { type FunctionArtifact, encodeArguments, getInitializer } from '@aztec/foundation/abi';
+
+import { type AuthWitnessProvider } from '../account/interface.js';
+import { type ContractArtifact } from '../api/abi.js';
+import { BaseContractInteraction, type SendMethodOptions } from '../contract/base_contract_interaction.js';
+import { Contract } from '../contract/contract.js';
+import { type ContractBase } from '../contract/contract_base.js';
+import { DeploySentTx } from '../contract/deploy_sent_tx.js';
+import { EntrypointPayload } from '../entrypoint/payload.js';
+import { type ContractInstanceWithAddress } from '../index.js';
+
+/**
+ * Options to pass to account contract initialization.
+ */
+type DeployAccountOptions = {
+  /** The Ethereum address of the Portal contract. */
+  portalContract?: EthAddress;
+  /** An optional salt value used to deterministically calculate the contract address. */
+  contractAddressSalt?: Fr;
+} & SendMethodOptions;
+
+/**
+ * Contract interaction for deploying an account contract. Handles fee preparation and contract initialization.
+ */
+export class DeployAccountMethod<TContract extends ContractBase = Contract> extends BaseContractInteraction {
+  private initializerArtifact?: FunctionArtifact;
+  private instance?: ContractInstanceWithAddress;
+
+  constructor(
+    pxe: PXE,
+    private authWitnessProvider: AuthWitnessProvider,
+    private publicKey: PublicKey,
+    private artifact: ContractArtifact,
+    private args: any[] = [],
+    private registerAccount: () => Promise<void>,
+    initializerNameOrArtifact?: string | FunctionArtifact,
+  ) {
+    super(pxe);
+    this.initializerArtifact = getInitializer(artifact, initializerNameOrArtifact);
+  }
+
+  /**
+   * Creates a TxExecutionRequest for deploying the contract.
+   * @param options - Prepares the contract for deployment by calling the request method and creating a TxExecutionRequest.
+   * @returns The TxExecutionRequest for deploying the contract.
+   */
+  public async create(options: DeployAccountOptions = {}): Promise<TxExecutionRequest> {
+    if (this.txRequest) {
+      return this.txRequest;
+    }
+
+    if (!this.initializerArtifact) {
+      throw new Error('Account contract can not be initialized without an initializer');
+    }
```

Comment:
Let's extend this to support account contracts _without_ initializers, in which case we should just publicly deploy the instance.